docs/advanced-usage/uuid.md

```diff
@@ -86,6 +104,83 @@ If you want all the role/permission objects to have a UUID instead of an integer
 - OPTIONAL: If you changed the field name in your migrations, you must set `protected $primaryKey = 'uuid';` to match.
 - Usually for UUID you will also set `public $incrementing = false;`. Remove it if it causes problems for you.
 
+Examples:
+
+Create new models:
+```bash
+php artisan make:model Role
+php artisan make:model Permission
+```
+
+`App\Model\Role.php`
+```php
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Concerns\HasUuids;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Spatie\Permission\Models\Role as SpatieRole;
+
+class Role extends SpatieRole
+{
+    use HasFactory, HasUuids;
+    protected $primaryKey = 'id';
+    protected $keyType = 'string';
+    public $incrementing = false;
+}
+```
+
+`App\Model\Permission.php`
+```php
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Concerns\HasUuids;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Spatie\Permission\Models\Permission as SpatiePermission;
+
+class Permission extends SpatiePermission
+{
+    use HasFactory, HasUuids;
+    protected $primaryKey = 'id';
```

Comment:
Also primary key `id` field on Role/Permission can be changed, 
Just use `protected $primaryKey = 'uuid';` on models, and update migration file, it is tested
https://github.com/spatie/laravel-permission/blob/9d7ea6da1d4ed1f41970823a3060aa459f8cae3a/tests/TestCase.php#L176-L179
https://github.com/spatie/laravel-permission/blob/9d7ea6da1d4ed1f41970823a3060aa459f8cae3a/tests/TestCase.php#L185-L188